providers/providers.go

```diff
@@ -193,8 +193,7 @@ func writeProviderEntry(dstore ds.Datastore, k cid.Cid, p peer.ID, t time.Time)
 
 func (pm *ProviderManager) gc() {
 	res, err := pm.dstore.Query(dsq.Query{
-		KeysOnly: true,
```

Comment:
ok, that's a subtle little change. What's the difference here?

Reply:
The expiration is in the value. We had this _before_ because we'd:

1. Determine that _all_ records have expired.
2. Delete _all_ records.

This caused us to remember a bunch of expired records indefinitely.

Reply:
OUCH!